src/devrules/notifications/channels/slack.py

```
@@ -0,0 +1,114 @@
+        self.channel_resolver = channel_resolver
+        self.channels_map = channels_map
+
+    def supports(self, event: NotificationEvent) -> bool:
+        return isinstance(event, DeployEvent)
+
+    def send(self, event: NotificationEvent) -> None:
+        with yaspin(text="Resolving channel...", color="green") as spinner:
+            try:
+                channel = self.channel_resolver(event, self.channels_map)
+            except ValueError as e:
+                raise RuntimeError(f"Failed to resolve Slack channel: {e}")
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)